src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/BannedAssembliesAnalyzer.cs

```diff
@@ -38,7 +38,10 @@ static void CheckType(ISymbolAnalysisContext context, ITypeSymbol type, ISymbol
                 {
                     if (BannedAssemblies.Contains(type.ContainingAssembly.Name))
                     {
-                        context.ReportDiagnostic(Diagnostic.Create(Descriptors.AZC0014, symbol.Locations.First(), BannedAssembliesMessageArgs), symbol);
+                        if (type.Name != "Utf8JsonReader" && type.Name != "Utf8JsonWriter")
+                        {
+                            context.ReportDiagnostic(Diagnostic.Create(Descriptors.AZC0014, symbol.Locations.First(), BannedAssembliesMessageArgs), symbol);
+                        } 
```

Comment:
this change reads like we are excluding the `Utf8JsonReader` and `Utf8JsonWriter` from the list of triggering this rule.
Is this expected?
I think we still need to report something when these two classes are used in public APIs, but not for those MRW related methods, such as `JsonModelWriteCore` and `JsonModelCreateCore`, but I did not see these two strings in your implementation.